pkg/loki/config_wrapper.go

```diff
@@ -147,13 +147,19 @@ func applyInstanceConfigs(r, defaults *ConfigWrapper) {
 // 1. Gives preference to any explicit ring config set. For instance, if the user explicitly configures Distributor's ring,
 // that config will prevail. This rule is enforced by the fact that the config file and command line args are parsed
 // again after the dynamic config has been applied, so will take higher precedence.
-// 2. If no explicit ring config is set, use the common ring configured if provided.
+// 2. If no explicit ring config is set, use the common ring configured if provided. If a common replication factor is given,
+// it will be reused by Ingester and IndexGateway ring.
 // 3. If no common ring was provided, use the memberlist config if provided.
 // 4. If no common ring or memberlist were provided, use the ingester's ring configuration.
 //
 // When using the ingester or common ring config, the loopback interface will be appended to the end of
-// the list of default interface names
+// the list of default interface names.
 func applyDynamicRingConfigs(r, defaults *ConfigWrapper) {
+	if r.Common.ReplicationFactor != defaults.Common.ReplicationFactor {
+		r.Ingester.LifecyclerConfig.RingConfig.ReplicationFactor = r.Common.ReplicationFactor
+		r.IndexGateway.Ring.ReplicationFactor = r.Common.ReplicationFactor
+	}
+
```

Comment:
Good call, I'll check how it will behave with this change.

Reply:
I pushed a commit addressing this. However,  I decided to create a new function (applyCommonReplicationFactor) dedicated to the replication factor, since `applyInstanceConfigs` isn't too related to it. WDYT?